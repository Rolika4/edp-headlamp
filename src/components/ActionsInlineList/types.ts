import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
