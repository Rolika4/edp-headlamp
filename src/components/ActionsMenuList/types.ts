import { KubeObjectAction } from '../../types/actions';

export interface ActionsMenuListProps {
  actions: KubeObjectAction[];
  handleCloseActionsMenu?(): void;
  anchorEl: HTMLElement | null;
}
