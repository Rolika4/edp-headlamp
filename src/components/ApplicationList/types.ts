import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface ApplicationListProps {
    applications: EDPCodebaseKubeObjectInterface[];
    onEdit?(): void;
    onDelete?(): void;
}
