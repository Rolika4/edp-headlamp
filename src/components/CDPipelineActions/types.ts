import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineActionsProps {
    kubeObjectData: EDPCDPipelineKubeObjectInterface;
    isDetailsPage?: boolean;
}
