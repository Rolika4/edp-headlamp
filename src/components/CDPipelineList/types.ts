import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineListProps {
    CDPipelines: EDPCDPipelineKubeObjectInterface[];
}
