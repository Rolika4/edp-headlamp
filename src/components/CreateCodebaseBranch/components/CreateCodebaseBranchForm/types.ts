import { EDPCodebaseKubeObjectInterface } from '../../../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../../../k8s/EDPCodebaseBranch/types';
import { DeepPartial } from '../../../../types/global';

export interface CreateCodebaseBranchFormProps {
    codebaseData: DeepPartial<EDPCodebaseKubeObjectInterface>;
    editorOpen: boolean;
    setEditorOpen(boolean): void;
    setDialogOpen(boolean): void;
    handleApply(
        newCodebaseBranchData: DeepPartial<EDPCodebaseBranchKubeObjectInterface>,
        newDefaultCodebaseBranchData?: DeepPartial<EDPCodebaseBranchKubeObjectInterface>
    ): void;
    isApplying: boolean;
}
