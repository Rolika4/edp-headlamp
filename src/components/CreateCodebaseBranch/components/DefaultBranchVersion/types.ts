import { CodebaseBranchCreationFormFieldInterface } from '../../types';

export interface DefaultBranchVersionProps extends CodebaseBranchCreationFormFieldInterface {}
