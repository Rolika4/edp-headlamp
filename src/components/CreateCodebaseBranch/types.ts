import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../k8s/EDPCodebaseBranch/types';
import { DeepPartial } from '../../types/global';

export interface CreateCodebaseBranchProps {
    codebaseData: DeepPartial<EDPCodebaseKubeObjectInterface>;
    open: boolean;
    onClose(): void;
    setOpen(boolean): void;
    handleApply(
        newCodebaseBranchData: DeepPartial<EDPCodebaseBranchKubeObjectInterface>,
        newDefaultCodebaseBranchData?: DeepPartial<EDPCodebaseBranchKubeObjectInterface>
    ): void;
    isApplying: boolean;
}
