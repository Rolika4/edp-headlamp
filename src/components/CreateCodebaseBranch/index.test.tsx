/**
 * @jest-environment jsdom
 */

import { describe } from '@jest/globals';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen, waitFor } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { CreateCodebaseBranch } from './index';
import { CreateCodebaseBranchProps } from './types';

describe('CreateCodebaseBranch', () => {
    it('should render correctly', async () => {
        const props: CreateCodebaseBranchProps = {
            codebaseData: {
                apiVersion: 'apiVersion',
                kind: 'Codebase',
                metadata: {
                    name: 'test-app-name',
                    namespace: 'test-namespace',
                },
                spec: {
                    defaultBranch: 'test-default-branch',
                    versioning: {
                        type: 'edp',
                        startFrom: '0.0.0-SNAPSHOT',
                    },
                },
            },
            open: true,
            setOpen: () => {},
            onClose: () => {},
            handleApply: () => {},
            isApplying: false,
        };
        const store = configureStore({
            reducer: () => ({}),
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <CreateCodebaseBranch {...props} />
                </SnackbarProvider>
            </Provider>
        );
        await waitFor(() => {
            expect(screen.getByTestId('create-codebase-branch')).toMatchSnapshot();
        });
    });
});
