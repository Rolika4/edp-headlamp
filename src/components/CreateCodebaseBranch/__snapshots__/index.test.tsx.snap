// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCodebaseBranch should render correctly 1`] = `
<div
  class="makeStyles-dialog-1"
  data-testid="create-codebase-branch"
>
  <div
    class="makeStyles-dialogTitle-2"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Create new branch
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <span />
        </span>
        Edit YAML
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-dialogContent-3"
  >
    <div
      class="makeStyles-form-4"
    >
      <form>
        <div
          class="makeStyles-formInner-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    style="display: flex;"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <span
                          class="makeStyles-labelWrap-7"
                        >
                          <span
                            class="MuiTypography-root makeStyles-label-8 MuiTypography-body1"
                          >
                            Branch Name
                          </span>
                          <span />
                        </span>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        style="display: flex; align-items: flex-end;"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                        >
                          <div
                            class="MuiFormControl-root MuiTextField-root"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input MuiInput-input"
                                name="branchName"
                                placeholder="Enter Branch Name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    style="display: flex;"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <span
                          class="makeStyles-labelWrap-7"
                        >
                          <span
                            class="MuiTypography-root makeStyles-label-8 MuiTypography-body1"
                          >
                            From Commit Hash 
                          </span>
                          <span />
                        </span>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        style="display: flex; align-items: flex-end;"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                        >
                          <div
                            class="MuiFormControl-root MuiTextField-root"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input MuiInput-input"
                                name="fromCommit"
                                placeholder="Enter Commit"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-actions-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              undo changes
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            style="margin-left: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              apply
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
